src/api/spec/models/workflow_spec.rb

```diff
@@ -37,4 +51,59 @@
       end
     end
   end
+
+  describe '#valid' do
+    context 'with a supported step' do
+      it { expect(subject).to be_valid }
+    end
+
+    context 'with several supported steps' do
+      let(:yaml) do
+        { 'steps' => [{ 'branch_package' => { source_project: 'project',
+                                              source_package: 'package' } },
+                      { 'branch_package' => { source_project: 'project',
+                                              source_package: 'package' } }] }
+      end
+
+      it { expect(subject).to be_valid }
+    end
+
+    context 'with several unsupported steps' do
+      let(:yaml) do
+        { 'steps' => [{ 'unsupported_step' => {} },
+                      { 'branch_package' => {} }] }
+      end
+
+      it { expect { subject }.to raise_error("Invalid workflow step definition: 'unsupported_step' is not a supported step.") }
+    end
+  end
+
+  describe '#errors' do
+    context 'with a supported step' do
+      it { expect(subject.errors).to be_empty }
+    end
+
+    context 'with several supported steps' do
+      let(:yaml) do
+        { 'steps' => [{ 'branch_package' => { source_project: 'project',
+                                              source_package: 'package' } },
+                      { 'branch_package' => { source_project: 'project',
+                                              source_package: 'package' } }] }
+      end
+
+      it { expect(subject.errors).to be_empty }
+    end
+
+    context 'with several unsupported steps' do
+      let(:yaml) do
+        { 'steps' => [{ 'unsupported_step' => {} },
+                      { 'unsupported_step' => {} },
+                      { 'branch_package' => {} }] }
+      end
+
+      it 'has several errors' do
+        expect { subject }.to raise_error("Invalid workflow step definition: 'unsupported_step' is not a supported step.\n'unsupported_step' is not a supported step.")
```

Comment:
I believe this might solve your failing spec.

```suggestion
        expect { subject }.to raise_error(Token::Errors::InvalidWorkflowStepDefinition, "Invalid workflow step definition: 'unsupported_step' is not a supported step.\n'unsupported_step' is not a supported step.")
```